app/controllers/mixins/actions/provider_actions/mass_transform.rb

```diff
@@ -0,0 +1,56 @@
+module Mixins
+  module Actions
+    module ProviderActions
+      module MassTransform
+        def vm_transform_mass
+          dialog = Dialog.find_by(:label => 'Transform VM')
+          @right_cell_text = _("Transform VMs to RHV")
+          if params.key?(:id)
+            provider = ExtManagementSystem.find_by(:id => params[:id].to_i)
+            dialog_initialize(
+              dialog.resource_actions.first,
+              :header     => @right_cell_text,
+              :target_id  => provider.id,
+              :target_kls => Provider.name
+            )
+          else
+            provider = ExtManagementSystem.find_by(:type => 'ManageIQ::Providers::Redhat::InfraManager')
```

Comment:
Multiple Redhat providers can be added. The best is to expect provider id always passed in. You can hide the provider selection field (but with an id assigned) in the dialog if there is only one provider available.

Reply:
@bzwei Hiding/showing of provider field is possible, but it is out of scope of this patch. In this patch I only add a proper target for the dialog to prevent the error.

When the mass transform button is clicked in 'Compute > Infrastructure > Virtual Machines' page, there is no `id` passed. So I cannot expect that provider id is always passed in.

It doesn't matter if there are many Redhat providers added. The provider object here is needed only as a placeholder to be used as dialog target. `find_by` will return one of the providers and it is enough for this purpose.